### PR TITLE
configure CORS according to latest spec

### PR DESCRIPTION
**Pull request by sebastianbenz**

### CODE REVIEW

**Comment by sebastianbenz** [comment]
//cc @andreban

**Comment by cramforce** [comment]
LGTM

**Comment by sebastianbenz** [comment]
Thanks for the quick review.

File: middlewares/amp-access-cors.js [hunk 1]
```diff
@@ -15,16 +15,75 @@
...
 module.exports = function(req, res, next) {
-  if (req.url.startsWith('/amp-')) {
-    // In practice, Origin should be restricted to a few well-known domains.
-    var requestingOrigin = req.header('Origin');
+  if (req.url.indexOf('/api/') > -1) {
+    // check the origin
+    var requestingOrigin = req.headers['origin'];
     console.log('---- requesting origin: ', requestingOrigin);
-    if (requestingOrigin) {
+    if (isValidOrigin(requestingOrigin)) {
```
Comment (dvoytenko) [inline middlewares/amp-access-cors.js]: Should this `if` do `else {403}`?

Comment (dvoytenko) [inline middlewares/amp-access-cors.js]: I see the fix. Good enough. Although, ideally, we'd do "return" in "else".

Comment (sebastianbenz) [inline middlewares/amp-access-cors.js]: Yup - didn't add the return as this means testing on a localhost no longer works. Probably should make it more explicit and check CORS only when not running on localhost - or is there a better way?

Comment (dvoytenko) [inline middlewares/amp-access-cors.js]: @sebastianbenz You mean heroku -> localhost? Or localhost -> localhost? One option you can check if your request is to localhost, you can allow it. This is a kind of a quirky test for development mode. Or you can also have a switch: "devmode" where you explicitly allow localhost.

**Pull request merged**